tests/testthat/test-target_market_share.R

```diff
@@ -1143,3 +1143,24 @@ test_that("`target_market_share` outputs only positive values of `production`(#3
 
   expect_false(any(out$production < 0))
 })
+
+test_that("NA in technology share (ADO876)", {
+  ald <- tibble::tribble(
+    ~company_id, ~name_company, ~is_ultimate_owner,  ~sector,           ~technology, ~year, ~plant_location, ~production,   ~production_unit,  ~emission_factor,               ~emission_factor_unit, ~ald_timestamp,
+    38998,  "parc eolien de bel air",              FALSE, "power", "renewablescap",  2020,            "FR",      0, "MW",  0.74613363337134, "tonnes of CO2 per MWh",      "2021 Q1",
+    38998,  "parc eolien de bel air",              FALSE, "power", "renewablescap",  2021,            "FR",      1, "MW",  0.74613363337134, "tonnes of CO2 per MWh",      "2021 Q1"
+  )
```

Comment:
this can be replace with the convenient helper function: 
``` r 
ald <- fake_ald(year = c(2020, 2021), production = c(0,1))

scenario <- fake_scenario(year = c(2020, 2021))

out <- target_market_share(fake_matched(), ald, scenario, region_isos_stable)

...
```